pkg/apis/core/v1beta1/types_secretbinding.go

```diff
@@ -36,7 +36,7 @@ type SecretBinding struct {
 	// +optional
 	Quotas []corev1.ObjectReference `json:"quotas,omitempty" protobuf:"bytes,3,rep,name=quotas"`
 	// Provider defines the provider type of the SecretBinding.
-	// This field is immutable when the SecretBindingProviderValidation feature gate is enabled.
+	// This field is immutable.
 	// +optional
 	Provider *SecretBindingProvider `json:"provider,omitempty" protobuf:"bytes,4,opt,name=provider"`
```

Comment:
Now we always validate the `SecretBindingProvider` is not `nil`, so I guess optional here is left for backward compatibility?

Reply:
Yes, exactly. It can be still the case that there are (unused) SecretBindings with `provider.type=nil`. Currently it is still possible to update such SecretBinding with provider.type and use it afterwards.